src/main/messageExtractor/JSONExtractor.java

```diff
@@ -0,0 +1,31 @@
+package messageExtractor;
+
+import org.json.simple.parser.JSONParser;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.ParseException;
+
+public class JSONExtractor implements MessageExtractorInterface{
```

Comment:
Does it extract JSON? The way I see it, it parses the JSON and returns a PheripheralAction object.